lib/eve.js

```diff
@@ -0,0 +1,63 @@
+import { useState, useEffect } from 'preact/hooks'
+
+// throttle a function call to avoid calling it too much
+export const throttle = (fn, delay, t, call = () => ((t = 0), fn())) => () =>
+  t || (t = setTimeout(call, delay))
+
+// create an observable or event
+export const eve = (...args) => {
+  const subs = new Set()
+  const sub = (fn) => (subs.add(fn), () => subs.delete(fn))
```

Comment:
the `sub` function is to add a new function to the subscriber, it also create an `unsub` callback that you can use to remove the function from the list.
```js
const unsub = sub(value => console.log('my value'))
// when you don't need it anymore
unsub()
```
this also works well with react `useEffect` hook, more on that later